src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -100,8 +114,60 @@ class AmendmentTableImpl : public AmendmentTable
 void
 AmendmentTableImpl::addInitial ()
 {
-    // For each amendment this version supports, construct the AmendmentState object by calling
-    // addKnown. Set any vetoes or defaults. A pointer to the AmendmentState can be stashed
+    // For each amendment this version supports, construct the AmendmentState
+    // object by calling addKnown. Set any vetoes or defaults. A pointer to the
+    // AmendmentState can be stashed
+
+    std::set<std::pair<uint256, std::string>> toAdd;
+
+    for (auto const& _ : detail::PreEnabledAmendments)
+    {
+        toAdd.emplace (uint256 (_.first), _.second);
+    }
+
+    {
+        // add the amendments from the config file
+        auto const& section = getConfig ().section (SECTION_AMENDMENTS);
+        int const numExpectedToks = 2;
+        for (auto const& _ : section.lines ())
+        {
+            boost::tokenizer<> tokenizer (_);
+            int numToks = 0;
+            uint256 id;
+            std::string friendlyName;
+            for(auto const& curTok : tokenizer)
+            {
+                ++numToks;
+                if (numToks > numExpectedToks)
+                {
+                    continue;  // continue counting so can report number in the
+                               // error
+                }
+
+                if (numToks == 1)
+                    id.SetHex (curTok);
+                else
+                    friendlyName = curTok;
+            }
+
+            if (numToks != numExpectedToks)
+            {
+                std::string const errorMsg =
+                    (boost::format (
+                         "The %1% section in the config file expects %2% "
+                         "items. Found %3%") %
+                     SECTION_AMENDMENTS % numExpectedToks % numToks).str ();
+                throw std::runtime_error (errorMsg);
```

Comment:
Regardless of whether you change the inner `for` loop to be straight line code, the number of tokens is way less useful to a confused user than the actual text that was parsed (in the `_` variable).  It would probably be better to say something along the lines of "....  Found: '{text in `_`}'.